src/tests/ftest/security/container_security_acl.py

```diff
@@ -0,0 +1,176 @@
+        self.log.info("(4)==>Verify container permission tT, rw-property")
+        self.log.info(
+            "(4.1)Update container-acl %s, %s, permission_type: %s with %s",
+            user_type, self.current_user, permission_type, cont_permission)
+        self.setup_container_acl_and_permission(
+            user_type, self.current_user, permission_type, cont_permission)
+        self.log.info(
+            "(4.2)Verify container_attribute: read, expect: %s", expect_read)
+        self.verify_cont_rw_property("read", expect_read)
+        self.log.info(
+            "(4.3)Verify container_attribute: write, expect: %s", expect_write)
+        self.verify_cont_rw_property(
+            "write", expect_write, property_name, property_value)
+        self.log.info(
+            "(4.4)Verify container_attribute: read, expect: %s", expect_read)
+        self.verify_cont_rw_property("read", expect_read)
+
+        #(5)Verify container permissions aA, rw-acl
+        permission_type = "acl"
+        self.log.info("(5)Verify container permission aA, rw-acl ")
+        self.log.info(
+            "(5.1)Update container-acl %s, %s, permission_type: %s with %s",
+            user_type, self.current_user, permission_type, cont_permission)
+        expect = "pass"  #User who created the container has full acl access.
+        permission_acl = ""
+        self.setup_container_acl_and_permission(
+            user_type, self.current_user, permission_type, cont_permission)
+        self.log.info("(5.2)Verify container_acl: write, expect: %s", expect)
+        self.verify_cont_rw_acl(
+            "write", expect, secTestBase.acl_entry(
+                user_type, self.current_user, cont_permission))
+        self.log.info("(5.3)Verify container_acl: read, expect: %s", expect)
+        self.verify_cont_rw_acl("read", expect)
+
+        #(6)Verify container permission o, set-owner
+        self.log.info("(6)==>Verify container permission o, set-owner")
+        permission_type = "ownership"
+        expect = "deny"
+        if "w" in cont_permission:
+            expect = "pass"
+        self.log.info(
+            "(6.1)Update container-set ownership %s, %s, permission_type:"
+            " %s with %s", user_type, self.current_user, permission_type,
+            cont_permission)
+        self.setup_container_acl_and_permission(
+            user_type, self.current_user, permission_type, cont_permission)
+        self.log.info("(6.2)Verify container_ownership: write, expect: %s",
+            expect)
+        self.verify_cont_set_owner(expect, test_user+"@", test_group+"@")
+        self.log.info("(6.3)Verify container_ownership: read, expect: %s",
+            expect)
+        self.verify_cont_rw_property("read", expect)
+
+        #(7)Verify container permission d, delete
+        self.log.info(" ")
+        self.log.info("(7)==>Verify container d, delete ")
+        expect = "pass"
+        self.verify_cont_delete(expect)
```

Comment:
Delete is an interesting case. I think you are creating the pool with default permissions above (tcd or rw). If a user has "d" permission on the pool, they can delete any container in it. If you want to test the container delete permission, I would suggest creating the pool with "tc" permissions for the owner and group. Then they can only delete the container if they have "d" permission on the container.

So possible container delete cases are:
- Pool permission includes "d", container permission includes "d": ALLOWED
- Pool permission includes "d", container permission does NOT include "d": ALLOWED
- Pool permission does NOT include "d", container permission includes "d": ALLOWED
- Pool permission does NOT include "d", container permission does NOT include "d": DENIED

Reply:
Thanks for the comment.
I have added 4 tests for container delete at step(7),